iu_xray_data.py

```diff
@@ -0,0 +1,218 @@
+                    the words for the vocabulary
+                * transforms (str): data augmentation methods for input images
+    """
+
+    def __init__(self, hparams):
+        self._hparams = HParams(hparams, self.default_hparams())
+        self.img_root = self._hparams.img_root
+        self.transforms = self.build_transform(self._hparams.transforms)
+        self.text_root = self._hparams.text_root
+        self.vocab = Vocab(self._hparams.vocab_path)
+        self.pathologies = self._hparams.pathologies
+
+    def __len__(self):
+        return len(os.listdir(self.text_root))
+
+    def __iter__(self):
+        for file_name in os.listdir(self.text_root):
+            yield self.__getitem__(file_name)
+
+    def __getitem__(self, file_name):
+        json_name = osp.join(self.text_root, file_name)
+        with open(json_name, 'r') as f:
+            items = list(DataPack.deserialize(f.read()))
+
+        # Get image tensor
+        key = items[0].img_study_path
+        assert file_name.replace('.json', '') == key
+        img_path = osp.join(self.img_root, key) + '.png'
+        image_tensor = self.get_image(img_path, self.transforms)
+
+        # Get the label for tag classification
+        tags = items[3].content
+        tag_index = [self.pathologies.index(tag) for tag in tags\
+            if tag in self.pathologies]
+
+        label = torch.zeros(len(self.pathologies))
+        label[tag_index] = 1
+
+        # Get the findings
+        findings = items[1]
+        assert isinstance(findings, Findings)
+
+        if findings.content:
+            caption = findings.content
+        else:
+            caption = 'the lungs are normal.'
+
+        caption_token = list()
+        max_word_num = 0
+
+        for sentence in caption.split('. '):
+            sentence = sentence.replace('.', '').split()
+            if len(sentence) == 0 or len(sentence) == 1:
+                continue
+            tokens = self.vocab.map_tokens_to_ids_py(sentence).tolist()
+            tokens.append(self.vocab.eos_token_id)
+
+            max_word_num = max(max_word_num, len(tokens))
+            caption_token.append(tokens)
+        sentence_num = len(caption_token)
+
+        return image_tensor, label, caption_token, max_word_num, sentence_num
+
+    @staticmethod
+    def build_transform(tsfm_list):
+        t = []
+        for func, args in tsfm_list:
+            t.append(getattr(tfms, func)(**args))
+        return tfms.Compose(t)
+
+    def get_image(self, img_root, transforms):
+
+        # In this way, we can skip the ToPILImage in the data augmentations,
+        # speeding up the data loading
+        image = pil_loader(img_root)
+        image_tensor = transforms(image)
+        return image_tensor
+
+    @staticmethod
+    def default_hparams():
+        r"""Returns a dictionary of default hyperparameters.
+        See the specific subclasses for the details.
```

Comment:
Can you fix the docstring and explain what the configurations are ?

Reply:
Changed the docstring to be exactly the same as the MLC. Detailed descriptions are included in the class level docstring.